core/src/main/scala/org/http4s/SecureSession.scala

```diff
@@ -0,0 +1,9 @@
+package org.http4s
+
+import java.security.cert.X509Certificate
+
+final case class SecureSession(
+    sslSessionId: String,
+    cipherSuite: String,
+    keySize: Int,
+    X509Certificate: Array[X509Certificate])
```

Comment:
Do we want to expose this as an `Array` or translate it into something immutable?  I'm worried about a case class that has a mutable element, but also don't want us to spend time eagerly converting a collection that will often be ignored.

Reply:
Here I was just following the convention for servlets since it defines these parameters that way, but yeah, it would make sense to be immutable.  Is there another lazier convention that is preferred to express these parameters or their optionality?  Or should the array just become an immutable list in this existing case class?

Reply:
This is a case that would be nice to have an immutable array wrapper in the standard library or cats.  Just something that doesn't eagerly traverse this collection unless someone actually cares to read it.
The only standard thing I can think of that's lazy and would hide the immutability is a stdlib Stream.  Something like `Stream.range(0, array.length).map(i => array(i))`.  Whether that's actually even lighter than creating a list of just a few certificates, I don't know.

I guess be fine converting it to a `List` and not worrying about it, unless someone wants to argue that every microsecond is sacred here.

Reply:
I was thinking ahead about the future Blaze integration of the code at https://github.com/sebastianvoss/http4s/tree/add_ssl_request_attributes, and I noticed a slight discrepancy how servlets return the SSL session id and how that branch returned that same information.  The servlet defines the SSL sessionId as a `String`, but the blaze branch as an `Array[Byte]`, owing to the fact that blaze retrieves it directly from the engine session as follows:
```
sessionId = engine.getSession.getId
```
So we could pick a representation and convert to it or have separate representations for blaze and for servlets, which would imply client code changes if rehosting between servlets and blaze and could mean putting a representation of `SecureSession` in both the servlet project and the blaze server projects.  I am not certain if the String servlet SSL session id is the same as the SSL session id in a different form as I have not researched the details of that conversion at the moment.  As a result it might make sense to keep the SSL Session id as a String if we want to unify the models since it's more general.

Reply:
I think it should be the same representation.

I think the `Array[Byte]` might be the sensible representation.  The `SSLSession` seems to be the most server agnostic representation to me, and it's the servlet that's doing something weird.  Unfortunately, that brings us back to a mutable type on the case class, which we could get around with `scodec.bits.ByteVector`.  But that's Yet Another Wrapper.

If you or @balatbn have a strong opinion, I'll defer to that.  My preference is more for unification than for any specific option.

Reply:
Array[Byte] is fine. For Blaze, we can directly use the value. However, for servlet, we have to convert the hex string back to Array[Byte].